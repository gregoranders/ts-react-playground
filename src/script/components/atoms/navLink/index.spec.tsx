import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import * as TestSubject from '@atoms/navLink';

describe(`${TestSubject.NavLink.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  describe('snapshot', () => {
    it('active', (): void => {
      const snapshot = create(
        <MemoryRouter initialEntries={['/about']}>
          <TestComponent to="/about" className="green">
            test
          </TestComponent>
        </MemoryRouter>,
      );
      snapshot.toJSON();
      expect(snapshot).toMatchSnapshot();
    });

    it('inactive', (): void => {
      const snapshot = create(
        <MemoryRouter initialEntries={['/']}>
          <TestComponent to="/about" className="green">
            test
          </TestComponent>
        </MemoryRouter>,
      );
      snapshot.toJSON();
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('mount', () => {
    it('active', () => {
      const testSubject = mount(
        <MemoryRouter initialEntries={['/about']}>
          <TestComponent to="/about">test</TestComponent>
        </MemoryRouter>,
      );
      expect(testSubject.text()).toBe('test');
    });
  });
});
