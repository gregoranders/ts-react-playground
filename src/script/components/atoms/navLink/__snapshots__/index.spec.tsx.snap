// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink snapshot active 1`] = `
<a
  aria-current="page"
  className="w3-bar-item w3-button w3-mobile w3-hover-theme green w3-theme-d5"
  href="/about"
  onClick={[Function]}
  style={Object {}}
>
  test
</a>
`;

exports[`NavLink snapshot inactive 1`] = `
<a
  aria-current={null}
  className="w3-bar-item w3-button w3-mobile w3-hover-theme green"
  href="/about"
  onClick={[Function]}
>
  test
</a>
`;
